skopt/searchcv.py

```diff
@@ -673,59 +550,75 @@ def step(self, X, y, space_id, groups=None, n_jobs=1):
         # optimizer minimizes objective, hence provide negative score
         optimizer.tell(params, [-score for score in local_results])
 
-        # fit the best model if necessary
-        if self.refit:
-            self._fit_best_model(X, y)
-
     def fit(self, X, y=None, groups=None):
         """Run fit on the estimator with randomly drawn parameters.
 
         Parameters
         ----------
-        X : array-like, shape = [n_samples, n_features]
-            Training vector, where n_samples in the number of samples and
-            n_features is the number of features.
+        X : array-like or sparse matrix, shape = [n_samples, n_features]
+            The training input samples. Internally, it will be converted to
+            ``dtype=np.float32`` and if a sparse matrix is provided
```

Comment:
Conversion to np.float32 is usually done with a particular estimator that is used with BayesSearchCV, but not BayesSearchCV itself. So one can have something like 
```python
import numpy as np
from sklearn.base import BaseEstimator, TransformerMixin
from sklearn.pipeline import make_pipeline
from sklearn.linear_model import Lasso

from skopt import BayesSearchCV

# strings as input
X = ['a'*np.random.randint(1,10) for i in range(100)]
# len of string should be estimated output
y = [len(x) for x in X]

# transformer on strings that converts them to features
class MyTransformer(BaseEstimator, TransformerMixin):
    def fit(self, X, y=None):
        return self

    def transform(self, X, y=None):
        X = [[len(x)] for x in X]
        X = np.array(X)
        return X

estimator = make_pipeline(
    MyTransformer(),
    Lasso()
)

model = BayesSearchCV(
    estimator=estimator,
    search_spaces={
        'lasso__alpha': (0.1, 10.0)
    },
    n_iter=8
)

model.fit(X, y)
print(model.score(X, y))
```
where inputs are strings. So consider deleting the comment about conversion of data.